### PR TITLE
Update ZSH Completion Files

### PR DESCRIPTION
**Pull request by zeroSteiner**
This PR updates our included zsh completion files with the latest options and help text from msfvenom and msfconsole. Updates per the request in #12932.

## Verification

List the steps needed to make sure this thing works

- [x] Install `zsh` if you don't already have it / are not already using it
    - [x] Install the completions by adding the directory in which they reside to the `$fpath` environment variable
- [x] Check the completion suggestions for consistency with their respective tools
    - [x] `msfconsole`
    - [x] `msfvenom`

## Example Usage
*Note this displays the menu output from my zsh configuration which I believe is a non-default setting so it may not look the exact same with a default configuration.*
```
./msfconsole --ask
option
--ask                 -a  -- Ask before exiting Metasploit or accept 'exit -y'                                                                                                                                                                                                                                                                                           
-c                        -- Load the specified configuration file                                                                                                                                                                                                                                                                                                       
--defer-module-loads      -- Defer module loading unless explicitly asked                                                                                                                                                                                                                                                                                                
--environment         -E  -- Specify the database environment to load from the configuration                                                                                                                                                                                                                                                                             
--execute-command     -x  -- Execute the specified string as console commands                                                                                                                                                                                                                                                                                            
--help                -h  -- Show help text                                                                                                                                                                                                                                                                                                                              
--history-file        -H  -- Save command history to the specified file                                                                                                                                                                                                                                                                                                  
--migration-path      -M  -- Specify a directory containing additional DB migrations                                                                                                                                                                                                                                                                                     
--module-path         -m  -- Specifies an additional module search path                                                                                                                                                                                                                                                                                                  
--no-database         -n  -- Disable database support                                                                                                                                                                                                                                                                                                                    
--output              -o  -- Output to the specified file                                                                                                                                                                                                                                                                                                                
--plugin              -p  -- Load a plugin on startup                                                                                                                                                                                                                                                                                                                    
--quiet               -q  -- Do not print the banner on startup                                                                                                                                                                                                                                                                                                          
--real-readline       -L  -- Use the system Readline library instead of RbReadline                                                                                                                                                                                                                                                                                       
--resource            -r  -- Execute the specified resource file (- for stdin)                                                                                                                                                                                                                                                                                           
--version             -v  -- Show version                                                                                                                                                                                                                                                                                                                                
--yaml                -y  -- Specify a YAML file containing database settings ```

### CODE REVIEW

**Review by wvu — COMMENTED** [review]
I'm not sure this commit belongs in this PR? https://github.com/rapid7/metasploit-framework/pull/12948/commits/0459e05420eb4971147d4693fc9fd39e5bc256c3

**Comment by zeroSteiner** [comment]
Whoops yeah I'm leaking commits. Let me get this branch updated to remove that since it is very much not ready yet.

File: external/zsh/_msfconsole [hunk 1]
```diff
@@ -21,18 +21,20 @@
...
-  {-m,--module-path}"[Specifies an additional module search path]:search path:_files -/" \
+  {-m,--module-path}"[Load an additional module path]:module path:_files -/" \
   {-n,--no-database}"[Disable database support]" \
   {-o,--output}"[Output to the specified file]:output file" \
   {-p,--plugin}"[Load a plugin on startup]:plugin file:_files" \
   {-q,--quiet}"[Do not print the banner on startup]" \
   {-r,--resource}"[Execute the specified resource file (- for stdin)]:resource file:_files" \
   {-v,--version}"[Show version]" \
-  {-x,--execute-command}"[Execute the specified string as console commands]:commands" \
+  {-x,--execute-command}"[EExecute the specified console commands (use ; for multiples)]:commands" \
```
Comment (adfoster-r7) [inline external/zsh/_msfconsole]: TTypo? `EExecute`

Comment (adfoster-r7) [reply external/zsh/_msfconsole]: 👀

**Review by wvu — APPROVED** [review]

**Comment by wvu-r7** [comment]
# Release Notes

This updates the Zsh completions for `msfconsole` and `msfvenom`.

**Pull request merged**